roles/kubernetes/preinstall/tasks/0020-verify-settings.yml

```diff
@@ -5,6 +5,15 @@
       - ansible_version.full|version_compare('2.3.0', '>=')
   run_once: yes
 
+- name: Stop if either kube-master, kube-node or etcd is empty
```

Comment:
Will this create any issues when you run playbooks like upgrade / scale and you limit to either masters, nodes or etcd's ?

Reply:
Good question. I didn't think of it.
I'll check...

Reply:
It works like a charm. ansible is clever enough to keep groups and other settings intact while using --limit. If to thing carefully it should behave this way because while limiting to some hosts you may still need group and host vars from others. Or you may need to delegate a task to another host. So ansible can't touch your inventory.

```
2018-09-20 16:52:08 mgserjio: tmp$ cat i.ini
[kube-master]
master1 ansible_host=127.0.0.1

[etcd]
etcd1 ansible_host=127.0.0.1

[kube-node]
node1 ansible_host=127.0.0.1
node2 ansible_host=127.0.0.1
node3 ansible_host=127.0.0.1
node4 ansible_host=127.0.0.1
node5 ansible_host=127.0.0.1

[k8s-cluster:children]
kube-master
kube-node
2018-09-20 16:52:21 mgserjio: tmp$ cat task.yml
- hosts: k8s-cluster
  connection: local

  tasks:
    - name: Stop if either kube-master, kube-node, etcd is empty
      assert:
        that: groups.get('{{ item }}')
      with_items:
        - kube-master
        - kube-node
        - etcd

- hosts: kube-node
  connection: local

  tasks:
    - name: Stop if either kube-master, kube-node, etcd is empty
      assert:
        that: groups.get('{{ item }}')
      with_items:
        - kube-master
        - kube-node
        - etcd
2018-09-20 16:52:27 mgserjio: tmp$ ansible-playbook -i i.ini task.yml --limit master1

PLAY [k8s-cluster] **********************************************************************************************************************************************************************

TASK [Gathering Facts] ******************************************************************************************************************************************************************
ok: [master1]

TASK [Stop if either kube-master, kube-node, etcd is empty] *****************************************************************************************************************************
ok: [master1] => (item=kube-master) => {
    "changed": false,
    "item": "kube-master",
    "msg": "All assertions passed"
}
ok: [master1] => (item=kube-node) => {
    "changed": false,
    "item": "kube-node",
    "msg": "All assertions passed"
}
ok: [master1] => (item=etcd) => {
    "changed": false,
    "item": "etcd",
    "msg": "All assertions passed"
}

PLAY [kube-node] ************************************************************************************************************************************************************************
skipping: no hosts matched

PLAY RECAP ******************************************************************************************************************************************************************************
master1                    : ok=2    changed=0    unreachable=0    failed=0
```